services/common/src/components/projects/ProjectLinks.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { FC, useEffect, useState } from "react";
+import { getProject, getProjects } from "@mds/common/redux/selectors/projectSelectors";
+import { useSelector, useDispatch } from "react-redux";
+import { Field, change } from "redux-form";
+import ProjectLinksTable from "@mds/common/components/projects/ProjectLinksTable";
+import { ILinkedProject, IProject } from "@mds/common/interfaces";
+
+import { Button, Col, Row, Typography } from "antd";
+import { FORM, USER_ROLES, getProjectStatusDescription } from "@mds/common/constants";
+import { isProponent, userHasRole } from "@mds/common/redux/reducers/authenticationReducer";
+import {
+  createProjectLinks,
+  fetchProjectsByMine,
+} from "@mds/common/redux/actionCreators/projectActionCreator";
+import { dateSorter } from "@mds/common/redux/utils/helpers";
+import RenderMultiSelect from "../forms/RenderMultiSelect";
+
+interface ProjectLinksProps {
+  viewProject: (record: ILinkedProject) => string;
+  tableOnly?: boolean; // only show the table, no inputs
+}
+// outside of component to sneak past "hooks can't be rendered conditionally"
+const ProjectLinkInput = ({ unrelatedProjects = [], mineGuid, projectGuid }) => {
+  const dispatch = useDispatch();
+  const [currentSelection, setCurrentSelection] = useState([]);
+  const formName = FORM.ADD_EDIT_PROJECT_SUMMARY;
+  const fieldName = "linked-projects";
+
+  if (!projectGuid) {
+    return (
+      <Typography.Paragraph>
+        Please save this record first to add links to other project applications.
+      </Typography.Paragraph>
+    );
+  }
+
+  const transformUnrelatedProjects = (projects) => {
+    const unrelated = projects.sort(dateSorter("update_timestamp", false)).map((p) => ({
+      value: p.project_guid,
+      label: `${p.project_title} ${new Date(p.update_timestamp).toDateString()}`,
+    }));
+    return unrelated;
+  };
+
+  const addRelatedProjects = () => {
+    dispatch(createProjectLinks(mineGuid, projectGuid, currentSelection)).then(() => {
+      dispatch(change(formName, fieldName, []));
```

Comment:
I am somewhat baffled as to why sometimes change wants formName and sometimes it does not (it's not me choosing one overload or another). Perhaps inconsistent versioning across services.